tests/kademlia/testencoding.nim

```diff
@@ -0,0 +1,157 @@
+{.used.}
+
+# Nim-Libp2p
+# Copyright (c) 2023 Status Research & Development GmbH
+# Licensed under either of
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+# at your option.
+# This file may not be copied, modified, or distributed except according to
+# those terms.
+
+import unittest2
+import ../../libp2p/protobuf/minprotobuf
+import ../../libp2p/protocols/kademlia/protobuf
+import ../../libp2p/multiaddress
+import results
+
+suite "kademlia protobuffers":
+  test "record encode/decode":
+    let rec = Record(
+      key: Opt.some(@[1'u8, 2, 3]),
+      value: Opt.some(@[4'u8, 5, 6]),
+      timeReceived: Opt.some("2025-05-12T12:00:00Z"),
+    )
+    let encoded = rec.encode()
+    let decoded = Record.decode(encoded).get()
+    check:
+      decoded.key.get() == rec.key.get()
+      decoded.value.get() == rec.value.get()
+      decoded.timeReceived.get() == rec.timeReceived.get()
+
+  test "peer encode/decode":
+    let maddr = MultiAddress.init("/ip4/127.0.0.1/tcp/9000").tryGet()
+    let peer = Peer(id: @[1'u8, 2, 3], addrs: @[maddr], connection: connected)
+    let encoded = peer.encode()
+    let decodedRes = Peer.decode(initProtoBuffer(encoded.buffer))
+    check decodedRes.isSome()
```

Comment:
Maybe add a `doAssert` here so you don't continue in case `decodedRes.isNone`

Reply:
It will fail anyway with something like this:
```
        ...     /home/r/vacp2p/nim-libp2p/tests/kademlia/testencoding.nim(39, 27): Check failed: decodedRes.isSome()
        ...  ... ...
        ...     Unhandled defect: Trying to access value with err Result [ResultDefect]
        ...   [FAILED ] (  0.00s) peer encode/decode
```